onchain/rollups/test/foundry/consensus/authority/AuthorityHistoryPairFactory.t.sol

```diff
@@ -0,0 +1,183 @@
+        historyFactory = new HistoryFactory();
+        factory = new AuthorityHistoryPairFactory(
+            authorityFactory,
+            historyFactory
+        );
+    }
+
+    function testFactoryCreation() public {
+        vm.recordLogs();
+
+        factory = new AuthorityHistoryPairFactory(
+            authorityFactory,
+            historyFactory
+        );
+
+        Vm.Log[] memory entries = vm.getRecordedLogs();
+
+        uint256 numOfFactoryCreated;
+
+        for (uint256 i; i < entries.length; ++i) {
+            Vm.Log memory entry = entries[i];
+
+            if (
+                entry.emitter == address(factory) &&
+                entry.topics[0] == AuthorityHistoryPairFactoryCreated.selector
+            ) {
+                ++numOfFactoryCreated;
+
+                address a;
+                address b;
+
+                (a, b) = abi.decode(entry.data, (address, address));
+
+                assertEq(address(authorityFactory), a);
+                assertEq(address(historyFactory), b);
+            }
+        }
+
+        assertEq(numOfFactoryCreated, 1);
+
+        assertEq(
+            address(factory.getAuthorityFactory()),
+            address(authorityFactory)
+        );
+        assertEq(address(factory.getHistoryFactory()), address(historyFactory));
+    }
+
+    function testNewAuthorityHistoryPair(address _authorityOwner) public {
+        vm.assume(_authorityOwner != address(0));
+
+        vm.recordLogs();
+
+        (Authority authority, History history) = factory
+            .newAuthorityHistoryPair(_authorityOwner);
+
+        testNewAuthorityHistoryPairAux(_authorityOwner, authority, history);
+    }
+
+    function testNewAuthorityHistoryPairAux(
+        address _authorityOwner,
+        Authority _authority,
+        History _history
+    ) internal {
+        Vm.Log[] memory entries = vm.getRecordedLogs();
+
+        uint256 numOfAuthorityCreated;
+        uint256 numOfHistoryCreated;
+
+        for (uint256 i; i < entries.length; ++i) {
+            Vm.Log memory entry = entries[i];
+
+            if (
+                entry.emitter == address(authorityFactory) &&
+                entry.topics[0] == AuthorityCreated.selector
+            ) {
+                ++numOfAuthorityCreated;
+
+                address a;
+                address b;
+
+                (a, b) = abi.decode(entry.data, (address, address));
+
+                assertEq(address(factory), a);
+                assertEq(address(_authority), b);
+            }
+
+            if (
+                entry.emitter == address(historyFactory) &&
+                entry.topics[0] == HistoryCreated.selector
+            ) {
+                ++numOfHistoryCreated;
+
+                address a;
+                address b;
+
+                (a, b) = abi.decode(entry.data, (address, address));
+
+                assertEq(address(_authority), a);
+                assertEq(address(_history), b);
+            }
+        }
+
```

Comment:
Not sure if we should test for `NewHistory` and `OwnershipTransferred` events too?

Reply:
Good idea!